flang/include/flang/Optimizer/Builder/Runtime/RTBuilder.h

```diff
@@ -137,6 +137,13 @@ constexpr TypeBuilderFunc getModel<void *>() {
   };
 }
 template <>
+constexpr TypeBuilderFunc getModel<void (*)(int)>() {
+  return [](mlir::MLIRContext *context) -> mlir::Type {
+    return fir::LLVMPointerType::get(context,
```

Comment:
Thanks for taking a look, yeah I'll update it so that it is easier to read. But I don't think it matters. Since LLVM 17 there has been no support for typed pointers in LLVM IR: https://llvm.org/docs/OpaquePointers.html